### PR TITLE
Minimum PHP version of 7.2

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
I took the opportunity to see which older versions of packages actually worked on PHP 7.3 and update too

I think this is an important step if we are going to fix issues with PHP8

Relates to #475

### CODE REVIEW

File: src/Prophecy/Call/CallCenter.php [hunk 1]
```diff
@@ -62,16 +62,9 @@ public function __construct(StringUtil $util = null)
...
-            // Limit argument was introduced in PHP 5.4.0
-            $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3);
-        } elseif (defined('DEBUG_BACKTRACE_IGNORE_ARGS')) {
-            // DEBUG_BACKTRACE_IGNORE_ARGS was introduced in PHP 5.3.6
-            $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
-        } else {
-            $backtrace = debug_backtrace();
-        }
+        // Limit backtrace to last 3 calls as we don't use the rest
+        // Limit argument was introduced in PHP 5.4.0
```
Comment (stof) [inline src/Prophecy/Call/CallCenter.php:66]: this second line of comment can be dropped

File: tests/Doubler/Generator/ClassMirrorTest.php [hunk 2]
```diff
@@ -261,10 +262,11 @@ public function it_marks_final_methods_as_unextendable()
 
     /**
      * @test
-     * @expectedException Prophecy\Exception\InvalidArgumentException
      */
     public function it_throws_an_exception_if_interface_provided_instead_of_class()
     {
+        $this->expectException(\Prophecy\Exception\InvalidArgumentException::class);
```
Comment (stof) [inline tests/Doubler/Generator/ClassMirrorTest.php:268]: I suggest moving that just before the call in which we expect it, which avoids accepting exceptions triggered by the setup logic.

Comment (stof) [reply tests/Doubler/Generator/ClassMirrorTest.php:268]: and this could use a use statement.

File: src/Prophecy/Prophecy/MethodProphecy.php [hunk 3]
```diff
@@ -92,10 +92,7 @@ public function __construct(ObjectProphecy $objectProphecy, $methodName, $argume
 
                     case 'Traversable':
                     case 'Generator':
-                        // Remove eval() when minimum version >=5.5
-                        /** @var callable $generator */
-                        $generator = eval('return function () { yield; };');
-                        return $generator();
+                        return (function () { yield; })();
```
Comment (Taluu) [inline src/Prophecy/Prophecy/MethodProphecy.php:95]: we could use a static callable, nay ?

File: src/Prophecy/Prophecy/MethodProphecy.php [hunk 4]
```diff
@@ -203,13 +200,11 @@ public function willYield($items)
             ));
         }
 
-        // Remove eval() when minimum version >=5.5
-        /** @var callable $generator */
-        $generator = eval('return function() use ($items) {
+        $generator =  function() use ($items) {
```
Comment (Taluu) [inline src/Prophecy/Prophecy/MethodProphecy.php:203]: same here

Comment (ciaranmcnulty) [reply src/Prophecy/Prophecy/MethodProphecy.php:203]: I don't think we can, as it is later bound to a class

**Comment by ciaranmcnulty** [comment]
Not sure why Scrutinizer can't do the composer install - they suggest committing composer.lock to resolve it

File: src/Prophecy/Prophecy/MethodProphecy.php (hunk 4 above)
Comment (Taluu) [reply src/Prophecy/Prophecy/MethodProphecy.php:203]: Oh didn't see that, so I'm trusting your judgement. ;)

**Pull request merged**